src/main/java/net/coderbot/iris/mixin/lang/MixinTranslationStorage.java

```diff
@@ -0,0 +1,39 @@
+package net.coderbot.iris.mixin.lang;
+
+import net.coderbot.iris.Iris;
+import net.minecraft.client.resource.language.TranslationStorage;
+import net.minecraft.resource.Resource;
+import net.minecraft.util.Language;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Inject;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfo;
+import org.spongepowered.asm.mixin.injection.callback.LocalCapture;
+
+import java.io.InputStream;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+
+@Mixin(TranslationStorage.class)
+public class MixinTranslationStorage {
+
+    private static final String LANGUAGE_LOAD = "Lnet/minecraft/util/Language;load(Ljava/io/InputStream;Ljava/util/function/BiConsumer;)V";
+
+    @Inject(method = "load(Ljava/util/List;Ljava/util/Map;)V", at = @At(value = "INVOKE", target = LANGUAGE_LOAD), locals = LocalCapture.CAPTURE_FAILHARD)
+    private static void iris$addLangEntries(List resources, Map<String, String> translationMap, CallbackInfo ci, Iterator var2, Resource resource, InputStream inputStream) {
+        String fileName = resource.getId().getPath().substring(resource.getId().getPath().indexOf("/"));
+        String langCode = fileName.substring(1, fileName.lastIndexOf("."));
+        try {
+            Language.load(Iris.class.getResourceAsStream("/assets/iris/lang" + fileName), translationMap::put);
+            Map<String, Map<String, String>> langMap = Iris.getCurrentPack().getLangMap();
```

Comment:
This isn't really the best approach. You insert values into the language map from the current shaderpack, but when you load a different shaderpack, the values from the previous shaderpack remain. Instead, it would be better to just allow all queries against the translationMap to optionally access this secondary language map instead. Or, maybe have a secondary instance of TranslationStorage. There just needs to be a way that doesn't pollute the vanilla language map with values from the shaderpack language files.

Alternatively, this *could* work as long as you force a resource reload upon loading a different shaderpack, but that seems like a bit overkill at the moment